### PR TITLE
Ports Bluespace Navigation Gigabeacon (#44807)

### PR DESCRIPTION
**Pull request by Cenrus**
https://github.com/tgstation/tgstation/pull/44807

## About The Pull Request

This PR add "Bluespace Navigation Gigabeacon" that shuttles can use as a target for the jump computer.

You can customize locked status by multitool or name of beacons by multitool with unscrewed beacon.
If you, or emp, lock beacon you are still able see it in choice list like a "Locked" destination, but unable to jump to it.

## Why It's Good For The Game

Now any spacemen can construct their own Space station on any Z lvl (that not reserved, centcom, away mission) and dock (by custom dock) his ship to it.

:cl: Dennok
add: Bluespace Navigation Gigabeacon
add: Navigation computer now use trait blacklist to filter out z levels to where it can't place custom dock.
/:cl:

### CODE REVIEW

File: code/modules/research/techweb/all_nodes.dm [hunk 1]
```diff
@@ -228,6 +228,15 @@
 	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 2500)
 	export_price = 5000
 
+/datum/techweb_node/regulated_bluespace
+	id = "regulated_bluespace"
+	display_name = "Regulated Bluespace Research"
+	description = "Bluespace technology using stable and balanced procedures. Required by galactic convention for public use."
+	prereq_ids = list("base")
```
Comment (MarkSuckerberg) [inline code/modules/research/techweb/all_nodes.dm]: I think it should be under either basic bluespace research or applied bliespace research

File: code/modules/shuttle/navigation_computer.dm [hunk 2]
```diff
@@ -10,10 +10,11 @@
 	var/list/jumpto_ports = list() //hashset of ports to jump to and ignore for collision purposes
 	var/obj/docking_port/stationary/my_port //the custom docking port placed by this console
 	var/obj/docking_port/mobile/shuttle_port //the mobile docking port of the connected shuttle
+	var/list/locked_traits = list(ZTRAIT_RESERVED, ZTRAIT_CENTCOM, ZTRAIT_AWAY, ZTRAIT_REEBE) //traits forbided for custom docking
```
Comment (MarkSuckerberg) [inline code/modules/shuttle/navigation_computer.dm:13]: Just being picky, but it's forbidden, not forbided

File: code/modules/shuttle/spaceship_navigation_beacon.dm [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	else
+		icon_state = "core-open"
+
+/obj/machinery/spaceship_navigation_beacon/power_change()
+	. = ..()
+	update_icon()
+
+/obj/machinery/spaceship_navigation_beacon/multitool_act(mob/living/user, obj/item/multitool/I)
+	if(panel_open)
+		var/new_name = "Beacon_[input("Enter the custom name for this beacon", "It be Beacon ..your input..") as text]"
```
Comment (MarkSuckerberg) [inline code/modules/shuttle/spaceship_navigation_beacon.dm:44]: What exactly is the second string for? 
`"It be beacon ..your input.."`

File: code/modules/shuttle/spaceship_navigation_beacon.dm [hunk 4]
```diff
@@ -0,0 +1,63 @@
+/obj/item/circuitboard/machine/spaceship_navigation_beacon
+	name = "Bluespace Navigation Gigabeacon (Machine Board)"
+	build_path = /obj/machinery/spaceship_navigation_beacon
+	req_components = list()
```
Comment (MarkSuckerberg) [inline code/modules/shuttle/spaceship_navigation_beacon.dm:4]: Is it built without components intentionally?

Comment (Cenrus) [reply code/modules/shuttle/spaceship_navigation_beacon.dm:4]: yes.

File: code/modules/shuttle/spaceship_navigation_beacon.dm (hunk 3 above)
Comment (Cenrus) [reply code/modules/shuttle/spaceship_navigation_beacon.dm:44]: It's beckoning your input.
rename

**Comment by Cenrus** [comment]
Issues addressed

File: code/modules/shuttle/spaceship_navigation_beacon.dm [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	update_icon()
+
+/obj/machinery/spaceship_navigation_beacon/multitool_act(mob/living/user, obj/item/multitool/I)
+	if(panel_open)
+		var/new_name = "Beacon_[input("Enter the custom name for this beacon", "It be Beacon ..your input..") as text]"
+		if(new_name && Adjacent(user))
+			name = new_name
+			to_chat(user, "<span class='notice'>You change beacon name to [name].</span>")
+	else 
+		locked =!locked
```
Comment (ike709) [inline code/modules/shuttle/spaceship_navigation_beacon.dm]:
```suggestion
		locked = !locked
```

File: code/modules/shuttle/spaceship_navigation_beacon.dm [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+		to_chat(user, "<span class='notice'>You [locked ? "" : "un"]lock [src].</span>")
+	return TRUE
+
+/obj/machinery/spaceship_navigation_beacon/examine()
+	.=..()
```
Comment (ike709) [inline code/modules/shuttle/spaceship_navigation_beacon.dm]:
```suggestion
	. = ..()
```

File: code/modules/shuttle/navigation_computer.dm (hunk 2 above)
Comment (ike709) [reply code/modules/shuttle/navigation_computer.dm:13]: Honestly I think a list of whitelisted traits would be better tbh. Thoughts @qwertyquerty ?

**Review by ike709 — CHANGES_REQUESTED** [review]
Address the things.

**Comment by Cenrus** [comment]
Tested and working, awaiting qwertys input.

**Comment by ike709** [comment]
@qwertyquerty

**Comment by Cenrus** [comment]
The requested changes have been committed.

**Pull request merged**